astropy/coordinates/tests/test_frames.py

```diff
@@ -1676,3 +1676,49 @@ def test_frame_coord_comparison():
     error_msg = "Can only compare SkyCoord to Frame with data"
     with pytest.raises(ValueError, match=error_msg):
         frame == coord  # noqa: B015
+
+
+@pytest.mark.parametrize(
+    ["s1", "s2"],
+    (
+        ((1,), (1,)),
+        ((2,), (1,)),
+        ((1,), (2,)),
+        ((2,), (2,)),
+        ((2, 1), (1,)),
+        ((1,), (2, 1)),
+        ((2, 1), (1, 3)),
+    ),
+)
+def test_altaz_broadcast(s1, s2):
+    """Note: Regression test for #5982"""
+    where = EarthLocation.from_geodetic(lat=45 * u.deg, lon=30 * u.deg, height=0 * u.m)
+    time = Time(np.full(s1, 58000.0), format="mjd")
+    angle = np.full(s2, 45.0) * u.deg
+    result = AltAz(alt=angle, az=angle, obstime=time, location=where)
+    assert result.shape == np.broadcast_shapes(s1, s2)
+
+
+def test_transform_altaz_array_obstime():
+    """Note: Regression test for #12965"""
+    obstime = Time("2010-01-01T00:00:00")
+    location = EarthLocation(0 * u.deg, 0 * u.deg, 0 * u.m)
+
+    frame1 = AltAz(location=location, obstime=obstime)
+    coord1 = SkyCoord(alt=80 * u.deg, az=0 * u.deg, frame=frame1)
+
+    obstimes = obstime + np.linspace(0, 15, 50) * u.min
+    frame2 = AltAz(location=location, obstime=obstimes)
+    coord2 = SkyCoord(alt=coord1.alt, az=coord1.az, frame=frame2)
+    assert np.all(coord2.alt == 80 * u.deg)
```

Comment:
I'd perhaps add another `assert coord2.shape == (50,)` just so that it is clear that part of the test is to ensure the shapes broadcast correctly.

Reply:
Done.